tuf/repository_lib.py

```diff
@@ -1466,6 +1468,47 @@ def generate_targets_metadata(targets_directory, target_files, version,
 
 
 
+def _add_keys_to_keydict(keydict, keyids, repository_name='default'):
+  """
+  Non-public function that generates keys for the keyids listed and adds
+  them to a key dictionary.
+  """
+
+  for keyid in keyids:
+
+    # Do we have a duplicate?
+    if keyids.count(keyid) > 1:
+      raise securesystemslib.exceptions.Error('Same keyid listed twice:'
+          ' ' + keyid)
```

Comment:
This feels odd. We do not handle this exception anywhere, is this likely to happen in practice?
Furthermore, this is all data we manage. If duplicates really are an issue, should we just prune them? What are we expecting the 'user' to do when they see this exception?

Reply:
There is a test case about it :grin: 
In practice `add_verification_key` has a check to avoid duplicates.
For the rest of your questions, I don't have a strong opinion but I agree this does not seem to be the right place of the check. Unfortunately,  I cannot claim that I can track all possible data entry points for roledb  ... while this is the only "exit" point.  Maybe this is the historical reason for it. 
I would gladly remove the check and its test but I don't have a great insight on what to put instead. Nothing?